library/common.h

```diff
@@ -191,24 +199,40 @@ inline void mbedtls_xor(unsigned char *r, const unsigned char *a, const unsigned
         uint8x16_t x = veorq_u8(v1, v2);
         vst1q_u8(r + i, x);
     }
+    // This if statement helps some compilers (e.g., IAR) optimise out the byte-by-byte tail case
+    // where n is a constant multiple of 16.
+    // It makes no difference for others (e.g. recent gcc and clang) if n is a compile-time
+    // constant, and very little difference if n is not a compile-time constant.
```

Comment:
(v minor) Is there a reason to use multiple `//`s rather than a multi-line `/* ... */` comment?